scraper/mapharma/mapharma.py

```diff
@@ -78,26 +81,35 @@ def campagne_to_centre(pharmacy: dict, campagne: dict) -> dict:
     return centre
 
 
-def get_mapharma_opendata(client: httpx.Client = DEFAULT_CLIENT) -> dict:
-    base_url = 'https://mapharma.net/opendata/rdv'
-    result = dict()
+def get_mapharma_opendata(
+    client: httpx.Client = DEFAULT_CLIENT, 
+    opendata_url: str = MAPHARMA_OPEN_DATA_URL, 
+    opendata_url_fallback: str = MAPHARMA_OPEN_DATA_URL_FALLBACK) -> dict:
     try:
-        request = client.get(base_url, headers=MAPHARMA_HEADERS)
+        request = client.get(opendata_url, headers=MAPHARMA_HEADERS)
         request.raise_for_status()
         return request.json()
+    except TimeoutException as hex:
+        logger.warning(f"{opendata_url} timed out {hex}")
     except httpx.HTTPStatusError as hex:
-        logger.warning(f'{base_url} returned error {hex.response.status_code}')
+        logger.warning(f'{opendata_url} returned error {hex.response.status_code}')
     try:
-        with open(Path('data', 'output', 'mapharma_open_data.json'), 'r', encoding='utf8') as f:
-            result = json.load(f)
-    except IOError as ioex:
-        logger.warning('Reading mapharma_open_data.json returned error {ioex}')
-    return result
+        request = client.get(opendata_url_fallback, headers=MAPHARMA_HEADERS)
+        request.raise_for_status()
+        return request.json()
+    except httpx.TimeoutException as hex:
+        logger.warning(f"{opendata_url_fallback} timed out {hex}")
+    except httpx.HTTPStatusError as hex:
+        logger.warning(f'{opendata_url_fallback} returned error {hex.response.status_code}')
+    return None
 
-def get_pharmacy_and_campagne(id_campagne: int, id_type: int) -> [dict, dict]:
+def get_pharmacy_and_campagne(
+    id_campagne: int, 
+    id_type: int, 
+    opendata_file: str = MAPHARMA_OPEN_DATA_FILE) -> [dict, dict]:
     opendate = list
```

Comment:
Je pense qu'il y a une typo (et je ne suis pas sûr que l'initialisation par `list` fonctionne). Ça devrait plutôt être ça je pense.
```suggestion
    opendata = list() 
```